pkg/operator/k8sutil/deployment.go

```diff
@@ -73,6 +90,40 @@ func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deploym
 		}
 		if d.Status.ObservedGeneration != original.Status.ObservedGeneration && d.Status.UpdatedReplicas > 0 && d.Status.ReadyReplicas > 0 {
 			logger.Infof("finished waiting for updated deployment %s", d.Name)
+
+			// Now we check if we can go to the next daemon
+			actionToCheck := "continue"
+			for _, daemon := range daemonsToCheckList {
```

Comment:
Even if there was no change to the pod spec (and the pod wasn't restarted) I believe this will still be called. If that's the case, we could skip the "ok to continue" checks.